### PR TITLE
Upgrade openssh

### LINKED ISSUE #16: Document openssh patches
There are quite a few [patches](https://github.com/cloudposse/bastion/tree/3ab66fb94ab345ff82d2fb8898b97297e763acd7/patches/openssh) in the Dockerfile applied to [openssh](https://github.com/cloudposse/bastion/blob/3ab66fb94ab345ff82d2fb8898b97297e763acd7/Dockerfile#L38) before it is compiled.

Could these patches, their source and purpose be documented, please? As the bastion server security is very important, any changes to the openssh implementation should be documented so it can be audited.

Also, would the current patch set prevent upgrading the Dockerfile to use the current openssh version `V_7_6_P1`  instead of `V_7_4_P1` which is used [currently](https://github.com/cloudposse/bastion/blob/3ab66fb94ab345ff82d2fb8898b97297e763acd7/Dockerfile#L6)?

Thank you!

### LINKED ISSUE #19: Upgrade OpenSSH to Latest Release
## what
* Upgrade to latest release: https://github.com/openssh/openssh-portable/releases

## why
* OpenSSH is quite out of date

### PR DESCRIPTION
**Pull request by alebabai**
## what
- update openssh-portable version
- update own patches (`original-command.diff` and `obfuscate-version.diff`)
- sync patches from alpine (delete old, pull new)

## why
closes #19 
closes #16 

## references
* https://github.com/cloudposse/bastion/issues/16#issuecomment-374456769

### CODE REVIEW

**Comment by alebabai** [comment]
Compilation still fails for all of these cases:
- completely without patches
- with old alpine patches and updated own patches
- only with updated alpine patches
- with update alpine patches and updated own patches

**Comment by alebabai** [comment]
The problem is in `realpath` function.
this patch `bsd-compatible-realpath.diff` should be updated too

**Comment by osterman** [comment]
@alebabai should we add a `Makefile` target to download the current patches the way we did for the helm charts?

**Comment by osterman** [comment]
@alebabai if it makes it easier to maintain, you can put the patches into vendor folders so we know what's upstream and our own.

**Review by osterman — COMMENTED** [review]
See inline comments

**Comment by alebabai** [comment]
> should we add a Makefile target to download the current patches the way we did for the helm charts?

I don't think so, because some patches could require to be updated (e.g. bsd-compatible-realpath).

> if it makes it easier to maintain, you can put the patches into vendor folders so we know what's upstream and our own.

yeap, it's good idea

File: patches/openssh/cloudposse/bsd-compatible-realpath.diff [hunk 1]
```diff
@@ -0,0 +1,197 @@
+diff --git a/openbsd-compat/openbsd-compat.h b/openbsd-compat/openbsd-compat.h
```
Comment (osterman) [inline patches/openssh/cloudposse/bsd-compatible-realpath.diff]: This was an `alpine` patch https://git.alpinelinux.org/cgit/aports/tree/main/openssh

Comment (osterman) [reply patches/openssh/cloudposse/bsd-compatible-realpath.diff]: Move to appropriate vendor folder

**Review by osterman — CHANGES_REQUESTED** [review]
Move patch to appropriate vendor folder

**Comment by osterman** [comment]
Please add a `README.md` to the `patches` folder that reads something like this:


```
OpenSSH will not compile out-of-the-box on alpine. For this reason, we use the official patches found here:
- https://git.alpinelinux.org/cgit/aports/tree/main/openssh

We also add a couple of our own patches. 

One patch ensures we have `SSH_ORIGINAL_COMMAND` available during pam auth so we can send slack notifications.
https://github.com/cloudposse/bastion/blob/master/patches/openssh/original-command.diff

The other patch obscures the version of OpenSSH. We use this to hide the SSH version so it's not announced to port-scanners.
https://github.com/cloudposse/bastion/blob/master/patches/openssh/obfuscate-version.diff

When upgrading version of OpenSSH, the patches might need to be regenerated. 


```

File: patches/openssh/cloudposse/bsd-compatible-realpath.diff (hunk 1 above)
Comment (alebabai) [reply patches/openssh/cloudposse/bsd-compatible-realpath.diff]: yeap, but i modified it - now it differs significantly from alpine one

**Review by osterman — APPROVED** [review]

**Comment by osterman** [comment]
Good job!

**Pull request merged**